balancer/weightedtarget/weightedtarget.go

```diff
@@ -154,9 +143,8 @@ func (b *weightedTargetBalancer) UpdateClientConnState(s balancer.ClientConnStat
 
 	b.targets = newConfig.Targets
 
-	if rebuildStateAndPicker {
-		b.stateAggregator.BuildAndUpdate()
-	}
```

Comment:
> This would mean that we will be sending a new picker even in the case where we only added new sub-balancers.

This particular case is desired, otherwise it can lead to inconsistent behavior.

Consider:

1. All current children are TRANSIENT_FAILURE
2. A new policy is added
3. State stays TRANSIENT_FAILURE under previous logic, because UpdateClientConnState never set the boolean.  This is wrong.
4. An update happens from one of the TF children (e.g. transition from TF to TF but a different picker)
5. NOW an update will happen that makes the state CONNECTING due to the policy added in (2)

We can also simplify by removing this bool.  Its purpose is completely superseded by the `Pause` && `Resume` functionality.

Aaaaaaaaaaaaaaaaaaand actually, you're really right we don't need this particular call, though.  It will force an update every time no matter what; instead the operations themselves should trigger the `Resume` to do the update.  So this should be removed.  But I think the example above is still useful to mention.

Reply:
SG. This `BuildAndUpdate` is redundant to the `BuildAndUpdate` made inside `Add()` and `Remove()`